pkgs/development/libraries/csfml/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchurl, cmake, sfml }:
+
+let
+  version = "2.4";
+in
+
+stdenv.mkDerivation rec {
+  name = "csfml-${version}";
+  src = fetchurl {
+    url = "https://github.com/SFML/CSFML/archive/${version}.tar.gz";
```

Comment:
Please use `fetchFromGitHub` for this kind of URLs: the archive may get regenerated, which may invalidate the hash.

Reply:
Is the preferred choice of `rev` for this a tag (in this case, `"${version}"`, or `"2.4"`) or a revision hash?

Reply:
I'm attempting to revise this patch to use `fetchFromGithub`, but I'm seeing the following error:

```
these derivations will be built:
  /nix/store/57si6znxrp5r6s81m50mxj5pddm414hh-source.drv
  /nix/store/7pkcj3k8ln1zhdpj5ph1599wmb0p93hz-csfml-2.4.drv
building path(s) ‘/nix/store/hr9zs38gy9axf3jqxl49d6580md36wil-source’

trying https://github.com/SFML/CSFML/archive/b5facb85d13bff451a5fd2d088a97472a685576c.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   152    0   152    0     0    152      0 --:--:-- --:--:-- --:--:--   894
100  123k  100  123k    0     0   123k      0  0:00:01 --:--:--  0:00:01 1233k
unpacking source archive /tmp/nix-build-source.drv-0/b5facb85d13bff451a5fd2d088a97472a685576c.tar.gz
output path ‘/nix/store/hr9zs38gy9axf3jqxl49d6580md36wil-source’ has r:sha256 hash ‘1q716gd7c7jlxzwpq5z4rjj5lsrn71ql2djphccdf9jannllqizn’ when ‘01cj7fzbdi8xsmwfdawwd00x81pv94fzmm6pqgp9d3973f3zrh92’ was expected
cannot build derivation ‘/nix/store/7pkcj3k8ln1zhdpj5ph1599wmb0p93hz-csfml-2.4.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/7pkcj3k8ln1zhdpj5ph1599wmb0p93hz-csfml-2.4.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

The hash it mentions not matching (`01cj7fzbdi8xsmwfdawwd00x81pv94fzmm6pqgp9d3973f3zrh92`) doesn't appear anywhere in my package, so I'm confused about what I've screwed up. I assume that it's a problem with the `sha256sum` value I added (`22c1fc871b278d96eec3d7d4fa1d49fb06d401689cabe678d51dc5b6be3b9205`, as obtained by running `wget http://github.com/SFML/CSFML/archive/b5facb85d13bff451a5fd2d088a97472a685576c.tar.gz && sha256sum b5facb85d13bff451a5fd2d088a97472a685576c.tar.gz`), but I don't understand what could have gone wrong.